experimental/lib/Support/BooleanLogic/Lexer.cpp

```diff
@@ -0,0 +1,110 @@
+//===- Lexer.cpp - Lexer for boolean expressions ----------------*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the LexicalAnalyzer class for tokenizing a string
+// representing a boolean logic expression. The constructor processes the input
+// string, identifying tokens and storing them in a list. The GetToken method
+// retrieves the next token, while peek allows looking ahead in the token
+// stream.
+//
+//===----------------------------------------------------------------------===//
+
+#include "experimental/Support/BooleanLogic/Lexer.h"
+#include "llvm/Support/raw_ostream.h"
+#include <string>
+#include <utility>
+#include <vector>
+
+using namespace llvm;
+using namespace dynamatic::experimental::boolean;
+
+// Function for reporting syntax errors encountered during lexical analysis.
+void dynamatic::experimental::boolean::syntaxError() {
+  llvm::errs() << "Syntax error in expression!\n";
+}
+
+LexicalAnalyzer::LexicalAnalyzer(std::string exp) {

```

Comment:
Make the argument a `StringRef`, which will remove the need for `std::move`. It's also better to initialize `expression` with the "special compiler syntax".

```cpp
LexicalAnalyzer::LexicalAnalyzer(StringRef exp) : expression(exp) {}
```